src/main/java/org/broadinstitute/dsm/route/BSPKitQueryRoute.java

```diff
@@ -46,11 +45,11 @@ public Object handle(Request request, Response response) throws Exception {
         }
 
         return TransactionWrapper.inTransaction(conn -> {
-            return getBSPKitInfo(conn, kitLabel, response);
+            return getBSPKitInfo(conn, kitLabel, response, this.notificationUtil);
         });
     }
 
-    public Object getBSPKitInfo(Connection conn, @NonNull String kitLabel, @NonNull Response response) {
+    public static Object getBSPKitInfo(Connection conn, @NonNull String kitLabel, @NonNull Response response, NotificationUtil notificationUtil) {
```

Comment:
making method `static` adds more complexity, for example, `notificationUtil` is instance field and you had to add it as a parameter which automatically adds more complexity, introducing `DAO` class would omit passing `Connection` to this method, method has several return points and it may return different type of results which adds more confusion, dissembling method into more specific parts would bring more cleaner code, except that, I always prefer to use method result for decisions, for example, you could use `null` result in upper method(`handle`) to decide whether make response status 400 or not, this approach would help to remove one more argument from method(Response response).